src/components/settings.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { Component } from 'react';
+import 'bootstrap/dist/css/bootstrap.css';
+import firebase from "firebase";
+import '../App.css'
+import Resizer from 'react-image-file-resizer'
+
+import {storage} from "../firebaseConfig";
+
+class Settings extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            username: "",
+            fName: '',
+            lName: '',
+            email: '',
+            bio: '',
+            signedIn: '',
+            url: '',
+            image: '', 
+            
+           
+        };
+        this.handleChange = this.handleChange.bind(this)
+    }
+    handleChange = e => {
+        let image = ''
+
+        if(e.target.files[0]) {
+            Resizer.imageFileResizer(
+                e.target.files[0],
+                125,
+                125,
+                'JPEG',
+                100,
+                0,
+                uri => {
+                    console.log(uri)
+                    
+                    firebase.database()
+                    .ref("users/" + localStorage.getItem('id')) 
+                    .update({  
+                            profileURL: uri,
+            
+                    });  
+                },
+                'base64'
+            );
+            image = e.target.files[0];
+            console.log(image);
+            
+        }
+        else {
+            console.log('ERROR')
+        }
+    }
+    handleChangeUserName(e) {
+        this.state.username = e.target.value
+        // set the changed state
+        this.setState({ username: this.state.username })
+
+
+    }
+    handleFirstName(e) {
+        this.state.fName = e.target.value
+        // set the changed state
+        this.setState({ fName: this.state.fName })
+    }
+    handleLastName(e) {
+        this.state.lName = e.target.value
+        // set the changed state
+        this.setState({ lName: this.state.lName })
+    }
+    handleEMAIL(e) {
+        this.state.email = e.target.value
+        // set the changed state
+        this.setState({ email: this.state.email })
+    }
+
+    handleBio(e) {
+        this.state.bio = e.target.value
+        // set the changed state
+        this.setState({ bio: this.state.bio })
+    }
+    
+
+    handleSubmit(e) {
+       
+        firebase
+            .database()
+            .ref("users/" + localStorage.getItem('id')) 
```

Comment:
use id from state. issue with usnig local storage is user could always delete their local storage if they wanted to which would cause our app to crash. we rely on it when necessary.